### PR TITLE
31 front authguards

### PR DESCRIPTION
**Pull request by clinche**

### CODE REVIEW

File: src/angularjs/src/app/auth-guard.service.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	if (!jwt_token)
+	{
+		console.log("[authGuardService] no token");
+		return false;
+	}
+
+	const res = await fetch('/api/auth/ft_verify?access_token=' + jwt_token,{
+		method: 'GET',
+		mode: 'cors'
+	});
```
Comment (clinche) [inline src/angularjs/src/app/auth-guard.service.ts:13]: we theorically do not need another route as only a post to /api/auth/profile should return a 200 if we correctly auth. we just need to put the token in authorization header

```js
await fetch('/api/auth/profile', {
   method: GET, 
   headers:  {
     'Authorization': 'Bearer ' + jwt_token 
   }
}
```

File: src/nestjs/src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -33,6 +33,21 @@ export class AuthService {
...
+		try {
+			await this.jwtService.verifyAsync(access_token);
+		}
+		catch(error)
+		{
+			return false;
+		}
+		return true;
+	}
+
```
Comment (clinche) [inline src/nestjs/src/modules/auth/auth.service.ts:50]: and so we shouldnt need that

File: src/angularjs/src/app/auth-guard.service.ts (hunk 1 above)
Comment (Pixailz) [reply src/angularjs/src/app/auth-guard.service.ts]: i know that there was a way to verify that's, was not sure were it was :p

File: src/nestjs/src/modules/auth/auth.service.ts (hunk 2 above)
Comment (Pixailz) [reply src/nestjs/src/modules/auth/auth.service.ts]: done

**Comment by Pixailz** [comment]
is it good to you @clinche ? as you are the author of this PR, you cannot review it

**Review by Pixailz — APPROVED** [review]
i approve those changes

**Review by Pixailz — APPROVED** [review]
again, i approve those change

**Comment by clinche** [comment]
I approve the approval

File: src/angularjs/src/app/components/header/header.component.ts [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss']
 })
 export class HeaderComponent {
-  userLoggedIn = true;
+	userLoggedIn = false;
+
+	async ngOnInit() {
+		this.userLoggedIn = await authGuardService();
```
Comment (clinche) [inline src/angularjs/src/app/components/header/header.component.ts:13]: isnt there a better way? just asking, i think its weird to have the auth guard called here and like that, but its maybe me

Comment (Pixailz) [reply src/angularjs/src/app/components/header/header.component.ts:13]: i'm not the best in angular, so yeah maybe it's a bit weird on how i implemented the `userLoggedIn` variable

but the authGuardService is basically a function that check if an user is logged in or not

**Pull request merged**